### PR TITLE
HTBHF-893 Add event logging for new card request.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java [hunk 1]
```diff
@@ -27,10 +30,18 @@ public void auditNewClaim(Claim claim) {
...
         NewClaimEvent newClaimEvent = NewClaimEvent.builder()
-                .claimantId(claim.getId())
+                .claimId(claim.getId())
                 .claimStatus(claim.getClaimStatus())
                 .eligibilityStatus(claim.getEligibilityStatus())
                 .build();
         eventLogger.logEvent(newClaimEvent);
     }
+
+    public void auditNewCard(UUID claimId, CardResponse cardResponse) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java]: @mikeSmith08 what do you think about removing the ClaimAuditor and just using the EventLogger directly? It doesn't seem to do very much anymore.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java [hunk 2]
```diff
@@ -27,10 +30,18 @@ public void auditNewClaim(Claim claim) {
...
+        NewCardEvent newCardEvent = NewCardEvent.builder()
+                .claimId(claimId)
+                .cardAccountId(cardResponse.getCardAccountId())
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java]: We should check for a null `CardResponse` as the new claim event logging does above.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java (hunk 1 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java]: I think it's still worthwhile as it nicely encapsulates the creating of the Event which the calling client then doesn't need to concern itself with.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java (hunk 2 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/audit/ClaimAuditor.java]: Done.

**Pull request merged**